### PR TITLE
fix: highlight code scroll console error (#7497)

### LINKED ISSUE #7497: Console error on scroll: Unable to preventDefault inside passive event listener due to target being treated as passive
OS: [windows, mac]
Browser: [chrome, edge]
Swagger-Editor version: 3.52.2
Swagger/OpenAPI version: [OpenAPI 3.0.3]

### Description of the issue
When scrolling "Example value" =  inside class .highlight-code the following error is thrown in the browser's console
`[Intervention] Unable to preventDefault inside passive event listener due to target being treated as passive. See https://www.chromestatus.com/features/6662647093133312`

### Steps to reproduce
[swagger.txt](https://github.com/swagger-api/swagger-ui/files/7159312/swagger.txt)
1. Copy past OpenAPI definition from the file above into https://editor.swagger.io/
2. collapse /pet endpoint
3. open console in the dev tools
4. scroll inside body -> example value  -> error will be shown in the console

### PR DESCRIPTION
**Pull request by kristina-kovacikova**
### Description

* fix for "Unable to preventDefault inside passive event listener due to target being treated as passive" 
  * commonly advised solution => setting CSS property `touch-action: none` did not fix the issue
  * as a workaround for the error manual event listener for scroll has been added to componentDidMount to be able to set option {passive:false} 

### Motivation and Context
Fixes #7497 

### How Has This Been Tested?

* go to `swagger-ui` and run `npm run dev` wait a while and open `localhost:3200`
  *   expand `PUT /pets` endpoint
  *   click inspect on the example body
    * in dev tools - elements search for `code` element within div with class "highlight-code"
      * edit `code` element as HTML and duplicate it => as a result the example body will be longer and scrollable
      * now that the example body is scrollable open `console in devtools` and scroll example body all the way till the whole page starts scrolling => there should be no errors in the console
      
    

### Screenshots (if appropriate):
image

### My PR contains... 

- [ ] No code changes (`src/` is unmodified: changes to documentation, CI, metadata, etc.)
- [ ] Dependency changes (any modification to dependencies in `package.json`)
- [x] Bug fixes (non-breaking change which fixes an issue)
- [ ] Improvements (misc. changes to existing features)
- [ ] Features (non-breaking change which adds functionality)

### My changes...
- [ ] are breaking changes to a public API (config options, System API, major UI change, etc).
- [ ] are breaking changes to a private API (Redux, component props, utility functions, etc.).
- [ ] are breaking changes to a developer API (npm script behavior changes, new dev system dependencies, etc).
- [x] are not breaking changes.

### Documentation
- [x] My changes do not require a change to the project documentation.
- [ ] My changes require a change to the project documentation.
- [ ] If yes to above: I have updated the documentation accordingly.

### Automated tests
- [x] My changes can not or do not need to be tested.
- [ ] My changes can and should be tested by unit and/or integration tests.
- [ ] If yes to above: I have added tests to cover my changes.
- [ ] If yes to above: I have taken care to cover edge cases in my tests.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by char0n** [comment]
@kristina-kovacikova thank you for the PR.

Can you please remove modified `package-lock.json` file from this PR? I assume the change was accidental.

File: src/core/components/highlight-code.jsx [hunk 1]
```diff
@@ -39,6 +39,16 @@ export default class HighlightCode extends Component {
     }
   }
 
+  componentDidMount() {
+    if (this.syntaxHighlighter) this.syntaxHighlighter.addEventListener("mousewheel", this.preventYScrollingBeyondElement, { passive: false })
```
Comment (char0n) [inline src/core/components/highlight-code.jsx:43]: What would you say if we would reduce the code here a little bit? We have almost the same two lines of code here.

```js
[this.syntaxHighlighter, this.pre]
  .filter(element => !!element)
  .map(element => element.addEventListener("mousewheel", this.preventYScrollingBeyondElement, { passive: false }));
```

Comment (kristina-kovacikova) [reply src/core/components/highlight-code.jsx]: thank you for the suggestion, I made the changes

File: src/core/components/highlight-code.jsx [hunk 2]
```diff
@@ -39,6 +39,18 @@ export default class HighlightCode extends Component {
     }
   }
 
+  componentDidMount() {
+    [this.syntaxHighlighter, this.pre]
+    .filter(element => !!element)
+    .map(element => element.addEventListener("mousewheel", this.preventYScrollingBeyondElement, { passive: false }))
```
Comment (mathis-m) [inline src/core/components/highlight-code.jsx:45]:
```suggestion
    .map(element => element?.addEventListener("mousewheel", this.preventYScrollingBeyondElement, { passive: false }))
```
Could be shortened one more time.

File: src/core/components/highlight-code.jsx [hunk 3]
```diff
@@ -39,6 +39,18 @@ export default class HighlightCode extends Component {
...
+  }
+
+  componentWillUnmount() {
+    [this.syntaxHighlighter, this.pre]
+    .filter(element => !!element)
+    .map(element => element.removeEventListener("mousewheel", this.preventYScrollingBeyondElement))
```
Comment (mathis-m) [inline src/core/components/highlight-code.jsx:51]:
```suggestion
    .map(element => element?.removeEventListener("mousewheel", this.preventYScrollingBeyondElement))
```
Could be shortened one more time.

**Review by mathis-m — CHANGES_REQUESTED** [review]

**Comment by kristina-kovacikova** [comment]
@mathis-m tested and committed your suggestions, thanks

File: src/core/components/highlight-code.jsx (hunk 3 above)
Comment (char0n) [reply src/core/components/highlight-code.jsx]: @mathis-m thanks for additional suggestions.

**Review by char0n — APPROVED** [review]
Great job! Thank you for contributing.

**Comment by char0n** [comment]
please build

**Pull request merged**